### PR TITLE
Add manual testing add edit commands

### PR DESCRIPTION
**Pull request by DennieDan**
I somehow forgot to merge this PR. Please help review.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -498,6 +498,70 @@ testers are expected to do more *exploratory* testing.
...
 
+### Adding a new student
+
+**Note:** You are encouraged to follow the test cases in order to see the expected results
+
+1. Adding a new student with all parameters specified
+
+   1. Simply type `add` into the command box and click `Enter`.
+
+   1. Copy the example shown together with the error message in the result display area.
```
Comment (teamotitan) [inline docs/DeveloperGuide.md:509]: Perhaps this part can be made a little clearer

I have a suggestion:

ii. You will see an error message in the result display area that teaches you how to use the command.
iii. In the error message, an example is provided as well. Copy the example.
iv. Test case(same as the example you copied): `add n/ John .....`



will there be expected result for this as well?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -498,6 +498,70 @@ testers are expected to do more *exploratory* testing.
...
+
+   1. Test case: `add n/John Doe p/98765432 e/johnd@example.com a/311, Clementi Ave 2, #02-25 t/Sunday-1000-1200 
+   s/Biology r/120 paid/6000 owed/3000`.
+
+1. Adding a new student with only compulsory parameters specified, order changed and case-insensitive command words
```
Comment (teamotitan) [inline docs/DeveloperGuide.md:514]: I have a suggestion:

"Adding a new student with these conditions: 1) only compulsory parameters are specified 2) order of parameters are changed 3) using both lowercase and uppercase letters for command word"

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -498,6 +498,70 @@ testers are expected to do more *exploratory* testing.
...
+
+   1. Test case: `add n/software Developer p/65894545 e/coderwithoutbug@gmail.com a/Jurong West Condo r/100 t/Sunday-1130-1330 s/PE`
+
+   1. Expected: The new student is successfully added with a warning message.
+
+1. Adding a new student with wrong prefixes
+
+   1. Test case: `add N/Teaching Assistant p/90908767 e/getaboveAtobeaTA123@hotmail.com a/21 Lower Kent Ridge Drive r/40.00 t/Thursday-1130-1330 s/Economics Paid/12.00`
+
+   2. Prefixes are not case-insensitive; therefore, an error message will be shown in result display.
```
Comment (teamotitan) [inline docs/DeveloperGuide.md:538]: Do you want to add Expected: ?

**Review by teamotitan — COMMENTED** [review]
I have a few suggestions. Again please if you feel maybe not that good no need to change

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (DennieDan) [reply docs/DeveloperGuide.md]: I have resolved it, please help review again. Thanks

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -498,6 +498,74 @@ testers are expected to do more *exploratory* testing.
...
+
+   1. Test case: `add n/software Developer p/65894545 e/coderwithoutbug@gmail.com a/Jurong West Condo r/100 t/Sunday-1130-1330 s/PE`
+
+   1. Expected: The new student is successfully added with a warning message.
+
+1. Adding a new student with mixed-case prefixes
+
+   1. Test case: `add N/Teaching Assistant p/90908767 e/getaboveAtobeaTA123@hotmail.com a/21 Lower Kent Ridge Drive r/40.00 t/Thursday-1130-1330 s/Economics Paid/12.00`
+
+   2. Expected: Prefixes are case-insensitive; therefore, a new contact with the corresponding information will be added to the end of the current list.
```
Comment (Soumyadip-cmd) [inline docs/DeveloperGuide.md:542]: Could consider leaving the front part about the prefixes as readers can infer from testcase and result

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -498,6 +498,74 @@ testers are expected to do more *exploratory* testing.
...
+### Editing an existing student
+
+1. Editing a student while all students are being shown
+
+   1. Prerequisites: List all students using the `list` command. Multiple students in the list.
+   
+   1. Test case: `eDit 3 owed/500 p/89873423`<br>Expected: Student number 3 will have the new values for their Owed Amount and Phone Number.
+   
+   1. Other incorrect edit commands to try: `edit`, `edit 0`, `edit x`, `...` (where x is larger than the list size)<br>
+   Expected: No student is deleted. Error details shown in the status message. Status bar remains the same.
```
Comment (Soumyadip-cmd) [inline docs/DeveloperGuide.md:553]: Perhaps can list out some incorrect commands like invalid index, editing name or phone leading to duplication, no parameters entered, etc.

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -498,6 +498,74 @@ testers are expected to do more *exploratory* testing.
...
+
+   1. Test case: `edit 1 n/Jonathan e/jonjon4343@canadian.com`
+   
+   1. Expected: The first student in the list's name and email will be updated. The list shows all students instead of the previous filtered list.
+
+1. Editing a student with invalid values
+
+   1. You can test the command using unacceptable values, e.g., `r/0`, `paid/100.000`, `n/   ` .
+   
+   1. Expected: Corresponding error messages will be displayed.
```
Comment (Soumyadip-cmd) [inline docs/DeveloperGuide.md:567]: Perhaps you can add separate testcases for the invalid values

**Review by Soumyadip-cmd — COMMENTED** [review]
Very interesting names and emails used. Just some considerations.

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -498,6 +498,77 @@ testers are expected to do more *exploratory* testing.
...
+   
+   1. Expected: The first student in the list's name and email will be updated. The list shows all students instead of the previous filtered list.
+
+1. Editing a student with invalid values
+
+   1. Test case: `edit 1 r/0` <br/>
+      Expected: An error message displayed reminds you that **RATE** must be from $0.01 to $1000.00.
+   
+   1. Test case: `edit 1 n/ p/654321234` <br/>
+      Expected: An error message displayed reminds you that **NAME** must not be empty. <br/>Notice that both values of **NAME** and **PHONE** are invalid the error message is shown in the order that the attributes shown in the app.
```
Comment (Feng1231) [inline docs/DeveloperGuide.md:569]: > Notice that both values of **NAME** and **PHONE** are invalid the error message is shown in the order that the attributes shown in the app.

Did you miss out on a fullstop or line break here?

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -498,6 +498,77 @@ testers are expected to do more *exploratory* testing.
...
+   1. Other incorrect edit commands to try: `edit`, `edit 0`, `edit x`, `...` (where x is larger than the list size)<br>
+   Expected: No student is deleted. Error details shown in the status message. Status bar remains the same.
+
+1. Editing a student in a filtered list
+
+   1. Prerequisites: Filter the students using `find` command.<br>Test case: `find n/irfan alex d/thursday friday`.
+
+   1. Test case: `edit 1 n/Jonathan e/jonjon4343@canadian.com`
+   
+   1. Expected: The first student in the list's name and email will be updated. The list shows all students instead of the previous filtered list.
```
Comment (Feng1231) [inline docs/DeveloperGuide.md:561]: The name and email of the first student in the list will be updated

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -498,6 +498,77 @@ testers are expected to do more *exploratory* testing.
...
+
+   1. Test case: `add n/a lucky Tester p/87903288 e/suchANiceApp@meta.sg a/COM1 B1 r/10 paid/0 owed/0 t/Tuesday-1500-1600
+   s/Physics`.
+
+   1. Expected: An error message shown: `This student already exists in the address book`. You can then refer [Finding a
+student]() to search for student details.
+
+1. Adding a new student with a clashing schedule
+
+   1. Test case: `add n/software Developer p/65894545 e/coderwithoutbug@gmail.com a/Jurong West Condo r/100 t/Sunday-1130-1330 s/PE`
```
Comment (Feng1231) [inline docs/DeveloperGuide.md:534]: Throws an error. subject does not include PE

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -498,6 +498,77 @@ testers are expected to do more *exploratory* testing.
...
+   
+   1. Expected: The student is successfully added.
+
+1. Adding a duplicated student
+
+   1. Test case: `add n/a lucky Tester p/87903288 e/suchANiceApp@meta.sg a/COM1 B1 r/10 paid/0 owed/0 t/Tuesday-1500-1600
+   s/Physics`.
+
+   1. Expected: An error message shown: `This student already exists in the address book`. You can then refer [Finding a
+student]() to search for student details.
```
Comment (Feng1231) [inline docs/DeveloperGuide.md:530]: you can then refer **to**
Also, is there supposed to be a link here?

**Pull request merged**